checkstyle-tester/diff.groovy

```diff
@@ -365,14 +365,72 @@ def cloneRepository(repoName, repoType, repoUrl, commitId, srcDir) {
         def lastCommitSha = getLastProjectCommitSha(repoType, srcDestinationDir)
         def commitIdSha = getCommitSha(commitId, repoType, srcDestinationDir)
         if (lastCommitSha != commitIdSha) {
-            def resetCmd = getResetCmd(repoType, commitId)
-            println "Resetting $repoType sources to commit '$commitId'"
-            executeCmd(resetCmd, new File("$srcDestinationDir"))
+            if (isGitSha(commitId)) {
+                // If commitId is a SHA, reset directly
+                def resetCmd = getResetCmd(repoType, commitId)
+                println "Resetting $repoType sources to commit '$commitId'"
+                executeCmd(resetCmd, new File("$srcDestinationDir"))
+            } else {
+                // If commitId is a branch or tag, fetch more data and then reset
+                fetchAdditionalData(repoType, srcDestinationDir, commitId)
+                def resetCmd = getResetCmd(repoType, commitId)
+                println "Resetting $repoType sources to commit '$commitId'"
+                executeCmd(resetCmd, new File("$srcDestinationDir"))
+            }
         }
     }
+
     println "$repoName is synchronized"
 }
 
+def getCloneCmd(repoType, repoUrl, srcDestinationDir) {
+    def cloneCmd = ''
+    switch (repoType) {
+        case 'git':
+            cloneCmd = "git clone $repoUrl $srcDestinationDir"
+            break
+        case 'hg':
+            cloneCmd = "hg clone $repoUrl $srcDestinationDir"
+            break
```

Comment:
Do we still use `hg`?

Reply:
please remove hg. it must be gone, lets forget about it

Reply:
It was removed. https://github.com/checkstyle/contribution/issues/822

Reply:
My bad. I have removed it. It might have stayed because of the confusion while reusing the existing PRs.